Document the Kubeflow Pipelines SDK for Tekton (#2341)
kubeflow/website: content/en/docs/pipelines/sdk/install-sdk.md

All of the SDK's classes and methods are described in the auto-generated [SDK reference docs](https://kubeflow-pipelines.readthedocs.io/en/latest/).

> **Note:** If you wan to run your pipelines on a [Tekton](https://tekton.dev/) backend

Reviewer: ```suggestion
**Note:** If you are using [Kubeflow Pipelines with Tekton backend]I(https://github.com/kubeflow/kfp-tekton), 
```

Reviewer 2: @animeshsingh nit:

```suggestion
**Note:** If you are using [Kubeflow Pipelines with the Tekton backend](https://github.com/kubeflow/kfp-tekton),
```

(I think an extra `I` found its way between `]` and `(` 😄 )

> instead of the default [Argo](https://argoproj.github.io/projects/argo) workflow

Reviewer: ```suggestion
instead of the default [Kubeflow Pipelines with Argo](https://github.com/kubeflow/pipelines) 
```

> engine, you can use the

Reviewer: ```suggestion
, you can use the
```
